### PR TITLE
Add `--overwrite` option for `config init` command

### PR DESCRIPTION
**Pull request by namanpatel1112**
Resolves https://github.com/zowe/zowe-cli/issues/1036

### CODE REVIEW

**Comment by zFernand0** [comment]
Some of these changes could be included in the #685 issue 😋

File: packages/imperative/__tests__/config/cmd/init/init.handler.test.ts
Comment (t1m0thyj) [reply packages/imperative/__tests__/config/cmd/init/init.handler.test.ts:197]: @namanpatel1112 The only difference I see between the expected and received object which is causing the test fail, is that the received object is missing the `$schema` property.

I believe the problem lies in the `initWithSchema` method. On line 78, the `$schema` property is added to the config JSON. Then on line 91, the config JSON is replaced with a new object that is missing the `$schema` property.
https://github.com/zowe/imperative/blob/05a05fdfac602175c5329b333368f68cb857b213/packages/imperative/src/config/cmd/init/init.handler.ts#L76-L93

Comment (namanpatel1112) [reply packages/imperative/__tests__/config/cmd/init/init.handler.test.ts]: The tests passed. I didn't have to swap the order of the writeFileSync expects. If I swapped them, the tests were failing. 🤔

File: packages/imperative/__tests__/config/cmd/init/init.handler.test.ts
Comment (awharn) [inline packages/imperative/__tests__/config/cmd/init/init.handler.test.ts:26]: Could you please remove this unused import?

File: packages/imperative/src/config/cmd/init/init.handler.ts
Comment (awharn) [inline packages/imperative/src/config/cmd/init/init.handler.ts:52]: Can you please remove the trailing space on this line?

File: packages/imperative/src/config/cmd/init/init.definition.ts
Comment (awharn) [inline packages/imperative/src/config/cmd/init/init.definition.ts:50]: I believe that we historically have the "ow" alias for overwrite operations. Would it make sense to add that alias here?

File: packages/imperative/src/config/cmd/init/init.definition.ts
Comment (awharn) [inline packages/imperative/src/config/cmd/init/init.definition.ts:55]: With overwrite commands, we often have a [confirmation option](https://github.com/zowe/imperative/blob/next/packages/imperative/src/config/cmd/auto-init/AutoInitConstants.ts#L81) to prevent people from invoking the option accidentally. In most places, this option is called "for-sure", with the alias "fs" for short. Would it make sense to add to this command as well? It could be potentially time consuming to create a configuration file again if overwrite is used.

**Review by awharn — COMMENTED** [review]
I have a couple of suggested changes and questions.

**Comment by zFernand0** [comment]
I would like to suggest one more thing.

Could we group the `--overwrite` tests (and probably the `--dry-run` tests as well) in their own `describe( )`?

The reason for asking for this is to make the `diff` a little easier to read through when reviewing the PR.

That said, I don't believe having a separate `describe( )` for each new test we add is necessarily the right thing to do either.

Personally, I don't feel too strongly as we could use tools other than `git diff` to review.

File: packages/imperative/__tests__/config/cmd/init/init.handler.test.ts
Comment (namanpatel1112) [reply packages/imperative/__tests__/config/cmd/init/init.handler.test.ts]: Removed it

File: packages/imperative/src/config/cmd/init/init.definition.ts
Comment (namanpatel1112) [reply packages/imperative/src/config/cmd/init/init.definition.ts]: Good point, I added a for-sure flag to go with it. :)

**Comment by namanpatel1112** [comment]
> I would like to suggest one more thing.
> 
> Could we group the `--overwrite` tests (and probably the `--dry-run` tests as well) in their own `describe( )`?
> 
> The reason for asking for this is to make the `diff` a little easier to read through when reviewing the PR.
> 
> That said, I don't believe having a separate `describe( )` for each new test we add is necessarily the right thing to do either.
> 
> Personally, I don't feel too strongly as we could use tools other than `git diff` to review.

So, should I leave them as they are right now? I found this diff checker: https://www.diffchecker.com/
But I am not sure if you had other tool(s) in mind.

File: packages/imperative/src/config/cmd/init/init.definition.ts
Comment (awharn) [inline packages/imperative/src/config/cmd/init/init.definition.ts:98]: Should this, and the example below, also include the --for-sure flag?

**Review by awharn — COMMENTED** [review]
One more small comment. Once this is fixed, this PR has my approval. 😃

File: packages/imperative/src/config/cmd/init/init.definition.ts
Comment (namanpatel1112) [reply packages/imperative/src/config/cmd/init/init.definition.ts]: Yes, I added it for both examples. Thanks for pointing it out.

**Review by awharn — APPROVED** [review]

**Review by awharn — COMMENTED** [review]
I just approved, but noticed that a lot of tests started failing because it states dry-run and overwrite are both specified. It looks like because both of those mutually exclusive options have default values, it believes they are both specified. It might be best to remove the default values on overwrite and dry-run.

File: packages/imperative/__tests__/config/cmd/init/init.handler.test.ts
Comment (t1m0thyj) [inline packages/imperative/__tests__/config/cmd/init/init.handler.test.ts:637]: Curious about the reason for this change? I think the `jest.resetAllMocks` call in the beforeEach block should mean no side effects of mocks are left behind for subsequent tests.

File: packages/imperative/__tests__/config/cmd/init/init.handler.test.ts
Comment (t1m0thyj) [inline packages/imperative/__tests__/config/cmd/init/init.handler.test.ts:686]: Again, curious about this change? 🙂

File: packages/imperative/src/config/cmd/init/init.handler.ts
Comment (namanpatel1112) [reply packages/imperative/src/config/cmd/init/init.handler.ts:124]: I made the requested change.

File: packages/imperative/__tests__/config/cmd/init/init.handler.test.ts
Comment (namanpatel1112) [reply packages/imperative/__tests__/config/cmd/init/init.handler.test.ts]: Good catch, this wasn't supposed to be removed. I added it back.

File: packages/imperative/__tests__/config/cmd/init/init.handler.test.ts
Comment (namanpatel1112) [reply packages/imperative/__tests__/config/cmd/init/init.handler.test.ts]: Again, I removed it unintentionally. I added it back.

**Review by t1m0thyj — APPROVED** [review]
LGTM!

**Review by zFernand0 — APPROVED** [review]

**Pull request merged**
